### DISCUSSION TITLE
Infinite Scroll without hitting the bottom of the list

### DISCUSSION THREAD

**Discussion by JerContact** [bosskmk/pluto_grid#815] [Q&A]
Currently it looks like there is only createFooter and createHeader options in PlutoGrid, is there a way to update based on the scroll itself to do infinite scrolling behind the scenes without any lag when someone gets to the bottom of the list?

**Comment by andzsinszan** [comment]
The infinite scroll demo 
https://weblaze.dev/pluto_grid/build/web/#feature/row-infinity-scroll
Source
https://github.com/bosskmk/pluto_grid/blob/master/demo/lib/screen/feature/row_infinity_scroll_screen.dart

The gist of it is the **PlutoInfinityScrollRows** class, which triggers a specified fetch callback function  (e.g., `doFetch(PlutoInfinityScrollRowsRequest req)`) when it require more data.
```dart
        createFooter: (s) => PlutoInfinityScrollRows(
          // First call the fetch function to determine whether to load the page.
          // Default is true.
          initialFetch: true,

          // Decide whether sorting will be handled by the server.
          // If false, handle sorting on the client side.
          // Default is true.
          fetchWithSorting: true,

          // Decide whether filtering is handled by the server.
          // If false, handle filtering on the client side.
          // Default is true.
          fetchWithFiltering: true,
          fetch: doFetch,
          stateManager: s,
        ),
```
---
_I changed `fetch:fetch` in the original code to `fetch:doFetch` above for clarity. To be honest, the naming `fetch` for the argument is probably not the best choice as it fails to convey that it requires a function pointer as value; not a boolean flag, nor an int (cf. `fetch: true`, `fetch: 100` may seem plausible setttings, but they are not). So probably `fetchFunction` or `fetchCallback` or `callback` would have been a better choice. One would not try to set a boolean or int for an argument named `fetchFunction`._
